### PR TITLE
Allow dashboard template vars with multiple options. Resolves #43

### LINKED ISSUE #43: Linter throws an error when dashboard has multiple choice variables
# Summary of the issue
When you have a multiple choice variable in your dashboard, linter will throw an error.

# Cause of the issue
`TemplateValue` struct is expecting 2 fields as string (https://github.com/grafana/dashboard-linter/blob/main/lint/lint.go#L35), but multiple choice variables are arrays:
```
  "templating": {
    "list": [
      {
        "current": {
          "selected": true,
          "text": [
            "All"
          ],
          "value": [
            "$__all"
          ]
        },
...
```

# Expected behaviour
Linter shouldn't throw an error when the dashboard has multiple choice variables.

# How to replicate
Just add a multiple choice variable to your dashboard. For example
```
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "iteration": 1649669111850,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "line",
            "fillOpacity": 0,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "none"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom"
        },
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "title": "Panel Title",
      "type": "timeseries"
    }
  ],
  "refresh": "",
  "schemaVersion": 35,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": [
      {
        "current": {
          "selected": true,
          "text": [
            "All"
          ],
          "value": [
            "$__all"
          ]
        },
        "hide": 0,
        "includeAll": true,
        "label": "Variable",
        "multi": true,
        "name": "query0",
        "options": [
          {
            "selected": true,
            "text": "All",
            "value": "$__all"
          },
          {
            "selected": false,
            "text": "foo",
            "value": "foo"
          },
          {
            "selected": false,
            "text": "bar",
            "value": "bar"
          }
        ],
        "query": "foo,bar",
        "queryValue": "",
        "skipUrlSync": false,
        "type": "custom"
      }
    ]
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "New dashboard",
  "version": 0,
  "weekStart": ""
}
```

### PR DESCRIPTION
**Pull request by rgeyer**
This PR does some unmarshal magic to allow either a string. or an array of strings for `text` and `value`.

The `TemplateValue` struct remains unchanged. If the value is an array, the first element is used for the value.

### CODE REVIEW

File: lint/lint.go [hunk 1]
```diff
@@ -83,6 +84,41 @@ func (t *Template) UnmarshalJSON(buf []byte) error {
...
+	} else {
+		switch tt := txt.(type) {
+		case string:
+			t.Text = txt.(string)
+		case []interface{}:
+			t.Text = txt.([]interface{})[0].(string)
+		default:
+			return fmt.Errorf("invalid type for field 'text': %v", tt)
+		}
+	}
```
Comment (tomwilkie) [inline lint/lint.go:104]:
```suggestion
	if txt, ok := raw["text"]; !ok {
		return errors.New("'text' property required")
	}
	
	switch tt := txt.(type) {
	case string:
		t.Text = txt.(string)
	case []interface{}:
		t.Text = txt.([]interface{})[0].(string)
	default:
		return fmt.Errorf("invalid type for field 'text': %v", tt)
	}
```

File: lint/lint.go [hunk 2]
```diff
@@ -83,6 +84,41 @@ func (t *Template) UnmarshalJSON(buf []byte) error {
...
+	} else {
+		switch vt := val.(type) {
+		case string:
+			t.Value = val.(string)
+		case []interface{}:
+			t.Value = val.([]interface{})[0].(string)
+		default:
+			return fmt.Errorf("invalid type for field 'value': %v", vt)
+		}
+	}
```
Comment (tomwilkie) [inline lint/lint.go:117]:
```suggestion
	if val, ok := raw["value"]; !ok {
		return errors.New("'value' property required")
	}

	switch vt := val.(type) {
	case string:
		t.Value = val.(string)
	case []interface{}:
		t.Value = val.([]interface{})[0].(string)
	default:
		return fmt.Errorf("invalid type for field 'value': %v", vt)
	}
```

**Review by tomwilkie — APPROVED** [review]
LGTM with minor nit

**Comment by rgeyer** [comment]
> LGTM with minor nit

Ahh right, I suppose I wouldn't need a conditional if I'm returning an error 😂

**Pull request merged**